tests/test_span_evaluator.py

```diff
@@ -392,3 +567,645 @@ def test_evaluate_with_custom_skipwords():
     assert result.pii_precision == pytest.approx(
         expected_precision
     ), f"Precision mismatch: expected {expected_precision}, got {result.pii_precision}"
+
+
+# ===== Component-Level Unit Tests =====
```

Comment:
added tests for the per_type metrics, error analysis, and more specific tests for some components in SpanEvaluator